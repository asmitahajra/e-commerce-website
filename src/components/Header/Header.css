.header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: lightgray;
    padding: 0.4rem;
}

.all-orders-and-cart {
    display:flex;
    align-items: center;
    justify-content: space-around;
}

.all-orders-and-cart >div,a,img {
    margin: 6px;
}

.cart {
    width: 5rem;
    height: 5rem;
}

.items-in-basket {
    display: flex;
    flex-direction: column;

}

.eshopper {
    font-size: 20px;
}

.items-in-basket {
    background-color: gray;
}