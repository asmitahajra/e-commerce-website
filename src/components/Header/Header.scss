$primary-text-color: black;

.margin-set {
    margin:0.4rem;
}

.header-light {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: lightgray;
    padding: 0.4rem;
    //padding: calc(6.4/16)rem;
    color: $primary-text-color;
    a {
        &:hover {
            color: cadetblue;
        }
        color: $primary-text-color;
    }
    .eshopper {
        font-weight: bold;
        font-size: 1.3rem;
    }
    .all-orders {
        font-size: 1.5rem;
    }
}

.header-dark {
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: black;
    padding: 0.4rem;
    //padding: calc(6.4/16)rem;
    color: white;
    a {
        color: white;
        &:hover {
            color: cadetblue;
        }
    }
    .eshopper {
        font-weight: bold;
        font-size: 1.3rem;
    }
    .all-orders {
        font-size: 1.5rem;
    }
}

.all-orders-and-cart {
    display:flex;
    align-items: center;
    justify-content: space-around;
}

.all-orders-and-cart {
    div,a,img{
        @extend .margin-set
    }
}

.cart {
    width: 5rem;
    height: 5rem;
}

.items-in-basket {
    font-size: 1.1rem;
    display: flex;
    flex-direction: column;

}

.eshopper {
    font-size: 1.1rem;
}

.items-in-basket {
    background-color: gray;
}