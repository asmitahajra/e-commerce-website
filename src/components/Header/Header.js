import React, { useContext } from 'react';
import './Header.scss';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { ThemeContext } from '../../ThemeContext';

const Header = ({ cartTotal }) => {
  const Theme = useContext(ThemeContext);
  const currentTextColor = Theme.color;
  return (
    <div className={currentTextColor === 'white' ? 'header-dark' : 'header-light'}>
      <Link className="eshopper" to="/"><b>E-Shopper</b></Link>
      <div className="all-orders-and-cart">
        <Link to="allOrders" className="all-orders">All Orders</Link>
        {/* <div className="go-to-cart"> */}
        <img data-testid="cart-img" className="cart" src="../assets/images/cart.jpeg" alt="cart-logo" />
        <div className="items-in-basket">
          <Link to="/cart">
            <div>
              My Basket
              <div>
                {cartTotal}
                {' '}
                items
              </div>
            </div>
          </Link>
        </div>
      </div>
    </div>
  );
};

Header.propTypes = {
  cartTotal: PropTypes.number.isRequired,
};
export default Header;
