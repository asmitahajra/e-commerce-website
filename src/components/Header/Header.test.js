/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from './Header';

describe(Header.name, () => {
  const mockProp = {
    cartTotal: 0,
  };
  afterEach(() => jest.clearAllMocks());

  test('should match snapshot', () => {
    const { container } = render(<BrowserRouter><Header {...mockProp} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });

  test('should open Homepage when E-Shopper link clicked', () => {
    render(<BrowserRouter><Header {...mockProp} /></BrowserRouter>);
    const eshopperLink = screen.getByText('E-Shopper');
    fireEvent.click(eshopperLink);
    expect(document.location.href).toBe('http://localhost/');
  });

  test('should open All Orders page when link clicked', () => {
    render(<BrowserRouter><Header {...mockProp} /></BrowserRouter>);
    const allOrdersLink = screen.getByText('All Orders');
    fireEvent.click(allOrdersLink);
    expect(document.location.href).toBe('http://localhost/allOrders');
  });

  test('should display product image', () => {
    render(<BrowserRouter><Header {...mockProp} /></BrowserRouter>);
    screen.getByTestId('cart-img');
  });

  test('should open Cart page on clicking My Basket and items', () => {
    render(<BrowserRouter><Header {...mockProp} /></BrowserRouter>);
    screen.getByText('My Basket');
    screen.getByText('0 items');
    const cartLink = screen.getByText('My Basket');
    fireEvent.click(cartLink);
    expect(document.location.href).toBe('http://localhost/cart');
    const itemsLink = screen.getByText('0 items');
    fireEvent.click(itemsLink);
    expect(document.location.href).toBe('http://localhost/cart');
  });

//   it('Should take to Home page on click', () => {
//     render(
//       <BrowserRouter>
//         <Header {...mockProp} />
//       </BrowserRouter>,
//     );
//     const logoLink = screen.getByTestId('img-tag');
//     fireEvent.click(logoLink);
//     expect(document.location.href).toBe('http://localhost/');
//   });
});
