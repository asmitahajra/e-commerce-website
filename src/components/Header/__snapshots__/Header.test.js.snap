// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div>
  <div
    class="header-light"
  >
    <a
      class="eshopper"
      href="/"
    >
      <b>
        E-Shopper
      </b>
    </a>
    <div
      class="all-orders-and-cart"
    >
      <a
        class="all-orders"
        href="/allOrders"
      >
        All Orders
      </a>
      <img
        alt="cart-logo"
        class="cart"
        data-testid="cart-img"
        src="../assets/images/cart.jpeg"
      />
      <div
        class="items-in-basket"
      >
        <a
          href="/cart"
        >
          <div>
            My Basket
            <div>
              0
               
              items
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
